### PR TITLE
feat(server): unique-by-corpus MCP server name with claude mcp add hint

### PR DESCRIPTION
**Pull request by yidakra**
## Summary

Each `dir2mcp up` instance now identifies itself with a unique MCP server name derived from the absolute path of the indexed directory: `dir2mcp-<slug>-<6-hex>`. Power users running many corpora side-by-side (e.g., `dir2mcp-stas-legal-a1b2c3` and `dir2mcp-research-notes-9f44ee`) can distinguish them in their MCP client list without manual bookkeeping.

The `up` banner — both foreground and daemon — now prints a ready-to-paste `claude mcp add --transport http <name> <url> ...` line using this name, with the protocol-version and (when applicable) Authorization header pre-filled.

## Why

Until now, every dir2mcp instance reported `serverInfo.name = \"dir2mcp\"` and the suggested registration alias was always the same. Anyone running >1 corpus had to invent unique aliases by hand, and the printed connection banner offered no help. A deterministic, path-keyed identifier solves that problem and is stable across reinstalls (same dir → same name).

## Shape

- **Auto-name:** `dir2mcp-<slug>-<6-hex>` where `<slug>` is the slugified basename of the indexed directory (lowercase, `[^a-z0-9]` → `-`, capped at 32 chars) and `<6-hex>` is the first 6 hex chars of `sha256(absRootDir)`.
- **Override:** `server.name: my-alias` in YAML, or `DIR2MCP_SERVER_NAME=my-alias` env. No CLI flag in v1; can be added later if needed.
- **Banner:** new \"Register with Claude\" section in both `printHumanConnection` (foreground) and `printDaemonReady` (daemon parent).
- **Server-reported name:** the two hardcoded `\"dir2mcp\"` literals at `internal/mcp/server.go:382` and `internal/mcp/transport_sdk.go:342` now read `s.cfg.ServerName`.

## Files

- `internal/identity/{name,name_test}.go` — pure helpers: `AutoServerName`, `Resolve`, `slugify`. Unit-tested for spaces, accents, all-symbol basenames, very long names, trailing slashes, determinism, and per-path distinctness.
- `internal/config/config.go` — new `Config.ServerName` field, YAML `server.name`, `DIR2MCP_SERVER_NAME` env, alias-map entry, `setServerStringFileScalar` case, `applyServerCoreFileParsed` branch, and `buildPersistedConfig` plumbing so the resolved name persists into the daemon snapshot.
- `internal/cli/up.go` — `resolveServerName(cfg)` helper called from the end of `applyScalarOverrides`, so by the time the server is constructed and the banner is printed, `cfg.ServerName` is already the override-or-auto-derived value.
- `internal/cli/up_daemon.go` — `printDaemonReady` signature changed to take `cfg` (replacing the previously-unused `stateDir` arg).
- `internal/cli/{registration_hint,registration_hint_test}.go` — new helper that renders the banner section with proper backslash continuations.
- `internal/mcp/server.go` — `NewServer` defensively auto-derives when `cfg.ServerName` is empty so test callers (which build minimal cfgs) don't have to wire identity.
- `tests/mcp/server_test.go` — initialize-handshake assertions for both override and auto-derive paths.
- `tests/cli/up_command_test.go` — banner integration test asserts `Register with Claude` + `claude mcp add --transport http dir2mcp-...` lines appear.
- `tests/security/cli_boundary_test.go` — new files added to the internal/cli ownership allow-list.
- `README.md` + `docs/SPEC.md` — new \"Server identity\" section + initialize-response note + `DIR2MCP_SERVER_NAME` row.
- Two unrelated single-line gofmt fixes (`tests/mcp/tools_test.go`, `tests/security/repo_split_boundary_test.go`) picked up by `make check`'s `gofmt -w`.

## Test plan

- [x] `go test ./internal/identity/...` (slug edge cases, determinism, distinctness)
- [x] `go test ./internal/cli/... -run \"TestBuildRegistrationCommand|TestPrintRegistrationHintSkipsWhenIncomplete\"`
- [x] `go test ./tests/mcp/... -run TestMCPInitialize_ServerName` (override + auto-derive cases over real HTTP handler)
- [x] `go test ./tests/cli/... -run TestUpBannerPrintsRegistrationHintWithUniqueName`
- [x] `make check` (full suite + lint + gofmt + python tests + build) green
- [ ] Manual smoke: `dir2mcp up` in a real corpus, copy/paste the printed `claude mcp add` line, confirm the new alias shows up in `claude mcp list`

## Notes

- Hash uses `filepath.Clean` of the absolute path, so `/var/lib/foo` and `/var/lib/foo/` yield the same name. Different parent dirs or different cases produce different names.
- The override path always wins; whitespace-only overrides are treated as empty and fall through to auto-derivation.
- This is non-breaking for the wire protocol (Claude Code routes by client-side alias, not by `serverInfo.name`).

## Summary by CodeRabbit

* **New Features**
  * Each `dir2mcp` instance now has a unique, auto-derived MCP server name based on the indexed directory
  * Server name can be overridden via the `DIR2MCP_SERVER_NAME` environment variable or YAML configuration
  * Added a "Register with Claude" banner that displays the recommended registration command for easier setup

* **Documentation**
  * Updated README and specification to document the new server identity feature and naming format

* **Tests**
  * Added comprehensive test coverage for the new server naming and registration features

[Review Change Stack](https://app.coderabbit.ai/change-stack/dirstral/dir2mcp/pull/184)

### CODE REVIEW

File: internal/cli/registration_hint.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// continuation. Lines that the user can paste verbatim use a trailing
+// backslash; the final line never does.
+func buildRegistrationCommand(name, url, protocolVersion string, requiresAuth bool) []string {
+	lines := []string{
+		fmt.Sprintf("claude mcp add --transport http %s %s \\", name, url),
+	}
+	headers := []string{}
+	if v := strings.TrimSpace(protocolVersion); v != "" {
+		headers = append(headers, fmt.Sprintf(`--header "MCP-Protocol-Version: %s"`, v))
+	}
```
[the remark this answers is not in the document]
Comment (yidakra) [reply internal/cli/registration_hint.go:43]: Fixed in f068ae4. Added shell-safe quoting via new `shellQuoteIfNeeded` helper so user-supplied overrides with spaces or single quotes still produce paste-safe commands. Auto-derived names (only [a-z0-9-]) stay bare. Tests cover spaces and embedded single quotes.

File: internal/config/config.go [hunk 2]
```diff
@@ -1502,6 +1516,9 @@ func applyEnvOverrides(cfg *Config, overrideEnv map[string]string) {
 	if cfg == nil {
 		return
 	}
+	if raw, ok := envLookup("DIR2MCP_SERVER_NAME", overrideEnv); ok && strings.TrimSpace(raw) != "" {
+		cfg.ServerName = strings.TrimSpace(raw)
+	}
```
[the remark this answers is not in the document]
Comment (yidakra) [reply internal/config/config.go]: Fixed in f068ae4. `DIR2MCP_SERVER_NAME` env now applies whenever the variable is *present* (regardless of whether the value is whitespace-only), so an empty/whitespace env can clear a YAML-set name and fall through to auto-derivation at `identity.Resolve` time. Behavior now matches what the PR description claimed.

**Comment by yidakra** [comment]
Addressed all reviewer items in `f068ae4`:

| Reviewer | Item | Fix |
|---|---|---|
| CodeRabbit (outside diff, major) | `marshalConfigYAML` doesn't write `server_name` → SaveFile/snapshot drops overrides | Added `writeScalar("server_name", cfg.ServerName)` next to `auth_mode` in `internal/config/config.go` |
| CodeRabbit (inline) | Daemon banner used raw `cfg.AuthMode` for auth-header hint | Already fixed by your direct commit `7b82940`; preserved in rebase |
| Copilot (inline) | `buildRegistrationCommand` didn't shell-quote name/URL | New `shellQuoteIfNeeded` helper applied in command build; tests for spaces + embedded single quotes |
| Copilot (inline) | `DIR2MCP_SERVER_NAME` env applied only when trimmed non-empty, contradicting PR description | Env now applies whenever set; whitespace-only values flow through to auto-derivation at `identity.Resolve` time |
| CodeRabbit (nitpick) | README `<folder>` vs `<slug>` inconsistency | Fixed `<folder>` → `<slug>` |
| CodeRabbit (nitpick) | Banner test only checked basename, not hash suffix | Now asserts the full `identity.AutoServerName` output (slug + 6-hex), with `EvalSymlinks` to canonicalize paths so the test matches macOS production cwd resolution |

`make check` passes locally.
